src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotUseInsecureDeserializerNetDataContractSerializerMethodsTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotUseInsecureDeserializerNetDataContractSerializerMethodsTests : DiagnosticAnalyzerTestBase
+    {
+        private static readonly DiagnosticDescriptor Rule = DoNotUseInsecureDeserializerNetDataContractSerializerMethods.RealMethodUsedDescriptor;
+
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        {
+            return new DoNotUseInsecureDeserializerNetDataContractSerializerMethods();
+        }
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new DoNotUseInsecureDeserializerNetDataContractSerializerMethods();
+        }
+
+        [Fact]
+        public void Deserialize_Diagnostic()
```

Comment:
Same question - We need a DFA trait either on test methods or test type. Doesn't seem like the `DiagnosticAnalyzerTestBase` has any default flow analysis trait.